app/tests/test_generic_api.py

```diff
@@ -274,10 +276,70 @@ def test_put_precondition(self):
                     f"/{STAC_BASE_V}/{endpoint}",
                     sample.get_json('put'),
                     content_type="application/json",
-                    HTTP_IF_MATCH=etag1
+                    HTTP_IF_MATCH=self.get_etag(endpoint)
                 )
                 self.assertStatusCode(200, response)
 
+    def test_wrong_media_type(self):
+        client_login(self.client)
+
+        for (request_methods, endpoint, data) in [
+            (
+                ['put', 'patch'],
+                f'collections/{self.collection["name"]}',
+                {},
+            ),
+            (
+                ['put', 'patch'],
+                f'collections/{self.collection["name"]}/items/{self.item["name"]}',
+                {},
+            ),
+            (
+                ['post'],
+                'collections',
+                self.factory.create_collection_sample(
+                    name=self.collection["name"],
+                    sample='collection-2',
+                ),
+            ),
+            (
+                ['post'],
+                f'collections/{self.collection["name"]}/items',
+                self.factory.create_item_sample(
+                    collection=self.collection.model,
+                    name=self.item["name"],
+                    sample='item-2',
+                ),
+            ),
+            (
+                ['post'],
+                f'collections/{self.collection["name"]}/items/{self.item["name"]}'
+                f'/assets',
+                self.factory.create_asset_sample(
+                    item=self.item.model,
+                    name=self.asset["name"],
+                    sample='asset-1-updated',
+                    media_type=self.asset['media_type'],
+                    checksum_multihash=self.asset["checksum_multihash"]
+                ),
+            ),
+            (['post'], 'search', {
+                "query": {
+                    "title": {
+                        "eq": "My item 1"
+                    }
+                }
+            }),
+        ]:
+            with self.subTest(endpoint=endpoint):
+                client_requests = [getattr(self.client, i) for i in request_methods]
```

Comment:
I like that :+1: 
Just for a better code readability, I would change the variable `i` to `method` as `i` is usually for index or number.